hudi-common/src/main/java/org/apache/hudi/common/util/RetryHelper.java

```diff
@@ -69,12 +69,12 @@ public RetryHelper(long maxRetryIntervalMs, int maxRetryNumbers, long initialRet
     this.taskInfo = taskInfo;
   }
 
-  public RetryHelper<T> tryWith(CheckedFunction<T> func) {
+  public RetryHelper<T, R> tryWith(CheckedFunction<T, R> func) {
     this.func = func;
     return this;
   }
 
-  public T start(CheckedFunction<T> func) throws IOException {
+  public <R extends Exception> T start(CheckedFunction<T, R> func) throws R {
```

Comment:
/nit `<R extends Exception>` here is redundant. `CheckedFunction` already declares `R`

Reply:
fix in the pr 5064